background/nudge/workflows.go

```diff
@@ -114,3 +117,39 @@ func (n *NudgeWorker) NotifyBehaviorOnRiskAreaWorkflow(ctx workflow.Context, acc
 
 	return nil
 }
+
+// HighRiskAccountFollowUpWorkflow is a workflow to follow up an account if it in risk by
+// it self reported symptoms
+func (n *NudgeWorker) HighRiskAccountFollowUpWorkflow(ctx workflow.Context, accountNumber string) error {
+	ctx = workflow.WithActivityOptions(ctx, activityOptions)
+
+	logger := workflow.GetLogger(ctx)
+
+	selector := workflow.NewSelector(ctx)
+
+	timerCancelCtx, _ := workflow.WithCancel(ctx)
+	timerFuture := workflow.NewTimer(timerCancelCtx, AccountHighRiskCheckInterval)
+	selector.AddFuture(timerFuture, func(f workflow.Future) {
+		logger.Info("Start periodically account self high risk nudge follow up")
+	})
+
+	selector.Select(ctx)
+
+	var shouldFollowUp bool
+	if err := workflow.ExecuteActivity(ctx, n.HighRiskAccountFollowUpActivity, accountNumber).Get(ctx, &shouldFollowUp); err != nil {
+		if err.Error() == background.ErrStopRenewWorkflow.Error() {
+			logger.Info("Stop following high risk for account", zap.Any("accountNumber", accountNumber))
+			return nil
+		}
+		logger.Error("Fail to check if an account needs to follow up", zap.Error(err))
+		return workflow.NewContinueAsNewError(ctx, n.HighRiskAccountFollowUpWorkflow, accountNumber)
+	}
+
+	if shouldFollowUp {
+		if err := workflow.ExecuteActivity(ctx, n.NotifyBehaviorFollowUpActivity, accountNumber).Get(ctx, &shouldFollowUp); err != nil {
+			logger.Error("Fail to send notification to follow up high risk user", zap.Error(err))
+		}
+	}
+
+	return workflow.NewContinueAsNewError(ctx, n.HighRiskAccountFollowUpWorkflow, accountNumber)
```

Comment:
```suggestion
	return nil
```

Reply:
And came across this: https://github.com/bitmark-inc/autonomy-api/blob/dcd8dc5c04666ffd023f905af12f13c6fe7a95d4/background/nudge/workflows.go#L58

Reply:
I do this because I would like the task to be run again. https://cadenceworkflow.io/docs/07_goclient/09_continueasnew

return nil will stop this workflow until it is triggered again.